includes/class-wcpdf-settings.php

```diff
@@ -693,6 +711,132 @@ public function get_sequential_number_store_method() {
 
 		return $method;		
 	}
+	
+	public function schedule_yearly_reset_numbers() {
+		if ( ! $this->maybe_schedule_yearly_reset_numbers() ) {
+			return;
+		}
+		
+		// checks AS functions existence
+		if ( ! function_exists( 'as_schedule_single_action' ) || ! function_exists( 'as_get_scheduled_actions' ) ) {
+			return;
+		}
+		
+		$next_year = strval( intval( current_time( 'Y' ) ) + 1 );
+		$datetime  = new \WC_DateTime( "{$next_year}-01-01 00:00:01", new \DateTimeZone( wc_timezone_string() ) );
+		$lock      = new Semaphore( $this->lock_name, $this->lock_time, array( wc_get_logger() ), $this->lock_context );
```

Comment:
@alexmigf I've not checked if `wc_get_logger()` has a `log()` method with parameters in the right order - I'll leave that to you (sorry, still on holiday!).

Reply:
`wc_get_logger()` returns a `WC_Logger` object, which indeed has a `log()` method: https://github.com/woocommerce/woocommerce/blob/b4502cf5d089791eaba0682d4d276d368e2c97bd/plugins/woocommerce/includes/class-wc-logger.php#L135-L161

When doing the `Semaphore` class implementation I notice that the arguments order wasn't ideal for the `WC_Logger` and fixed it for this plugin class, don't know if you want me to do anything more.

Reply:
@alexmigf Ah, I see. No problem.